command/agent.go

```diff
@@ -371,6 +371,7 @@ func (c *AgentCommand) Run(args []string) int {
 		// Check if a default namespace has been set
 		mountPath := config.AutoAuth.Method.MountPath
 		if cns := config.AutoAuth.Method.Namespace; cns != "" {
+			namespace = cns
```

Comment:
I wonder if we should incorporate the logic for EnvVaultNamespace here?  It's confusing to rely on the code in consult-template that consults the env when we don't specify a namespace explicitly.  Especially since we *do* look at the env to set mountPath.

Reply:
The EnvVaultNamespace logic here seems like it is only applicable to the auth subsystem of agent, and it seems like a sane thing to do. It is possible that we can refactor this to better manage the code, but I am feeling reluctant to touch it at this point in the release cycle.

Reply:
It looks like by setting `namespace = cns`, if there is an env var set, the templating code will ignore it.